src/main/java/sx/blah/discord/handle/impl/obj/Guild.java

```diff
@@ -779,6 +778,18 @@ public IEmoji getEmojiByName(String name) {
 	}
 
 	@Override
+	public IEmoji addEmoji(String name, BufferedImage image){
+		if (name.length() < 2 || name.length() > 32 || DiscordUtils.EMOJI_CREATE_NAME.matcher(name).find())
+			throw new DiscordException("Emoji names must be at least 2 characters long and can only contain alphanumeric characters and underscores.");
+
+		DiscordUtils.checkPermissions(client, this, EnumSet.of(Permissions.MANAGE_EMOJIS));
+
+		DataBufferByte data = (DataBufferByte) image.getRaster().getDataBuffer();
+		EmojiObject response = ((DiscordClientImpl) client).REQUESTS.POST.makeRequest(DiscordEndpoints.GUILDS + getID() + "/emojis", new EmojiCreateRequest(name, data.getData()), EmojiObject.class);
```

Comment:
We were talking about this in api-help, and we thought that it would just silently error which would make this function return null. The user would then try to use the returned IEmoji (currently null) which would cause an NPE. This would not be D4Js fault, but we should throw an exception saying that it was not completed successfully and that they need to get their bot whitelisted.

It would be nice if D4J catches those errors thrown by discord (in the returned json) and throws a DiscordException with the message from Discord on its own.